js/lib/ui.js

```diff
@@ -326,38 +326,7 @@ elgg.ui.loginHandler = function(hook, type, params, options) {
  * @requires jqueryui.datepicker
  */
 elgg.ui.initDatePicker = function() {
-	function loadDatePicker() {
-		$('.elgg-input-date').datepicker({
-			// ISO-8601
-			dateFormat: 'yy-mm-dd',
-			onSelect: function(dateText) {
-				if ($(this).is('.elgg-input-timestamp')) {
-					// convert to unix timestamp
-					var dateParts = dateText.split("-");
-					var timestamp = Date.UTC(dateParts[0], dateParts[1] - 1, dateParts[2]);
-					timestamp = timestamp / 1000;
-
-					var id = $(this).attr('id');
-					$('input[name="' + id + '"]').val(timestamp);
-				}
-			},
-			nextText: '&#xBB;',
-			prevText: '&#xAB;',
-			changeMonth: true,
-			changeYear: true
-		});
-	}
-
-	if (!$('.elgg-input-date').length) {
-		return;
-	}
-
-	// require language module if necessary
-	var deps = [];
-	if (elgg.get_language() != 'en') {
-		deps.push('jquery-ui/i18n/datepicker-'+ elgg.get_language() + '.min');
-	}
-	require(deps, loadDatePicker);
+	elgg.deprecated_notice('elgg.ui.initDatePicker has been deprecated. Use input/date AMD module instead');
```

Comment:
You saying it's not possible, or do you mean that synchronicity is part of
BC?
On Dec 15, 2015 1:10 AM, "Steve Clay" <notifications@github.com> wrote:

> In js/lib/ui.js
> <https://github.com/Elgg/Elgg/pull/9224#discussion_r47579776>:
>
> > -			changeMonth: true,
> > -			changeYear: true
> > -		});
> > -	}
> > -
> > -	if (!$('.elgg-input-date').length) {
> > -		return;
> > -	}
> > -
> > -	// require language module if necessary
> > -	var deps = [];
> > -	if (elgg.get_language() != 'en') {
> > -		deps.push('jquery-ui/i18n/datepicker-'+ elgg.get_language() + '.min');
> > -	}
> > -	require(deps, loadDatePicker);
> > +	elgg.deprecated_notice('elgg.ui.initDatePicker has been deprecated. Use input/date AMD module instead');
>
> For BC we can't replace synchronously defined functions with async
> modules, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9224/files#r47579776>.
>